exercises/test_x2.py

```diff
@@ -0,0 +1,67 @@
+import requests
+import pytest
+
+class TestKnockknockApi:
+    @pytest.fixture
+    def fixture(self):
+        return requests.get('http://localhost:8000/knockknock')
+
+    def test_status_code_is_ok(self, fixture):
+        assert fixture.status_code == 200
+    
+    def test_response_is_correct(self, fixture):
+        assert fixture.text == 'Who\'s there?'
```

Comment:
This can all be done in a single test:
- Fixtures are for setup/teardown. You're testing the `GET` call, it should be in the test.
- Asserts on both status code and response body can go in the same test. It's good for a test to test one thing only. Since making an API call is relatively expensive, in API testing that "one thing" is for example a `GET` call like it is here. So in a single test you might check the status code, check some headers, do schema validation on the response, and check some specific values in the response.

Reply:
Using fixtures in this way does avoid some code duplication. We'll solve that in a different way in exercise 4: by building an API client that the tests will call instead of using the `requests` library directly.